btk/measure.py

```diff
@@ -1,175 +1,239 @@
+
+            # make sure output is in the correct format.
+            if not isinstance(out["catalog"], astropy.table.table.Table):
+                raise TypeError(
+                    "The output dictionary of at least one of your measurement functions does not"
+                    "contain an astropy table as the value of the key 'catalog'."
+                )
+
+            if isinstance(batch["blend_images"], np.ndarray):
+                if not ("dx" in out["catalog"].colnames and "dy" in out["catalog"].colnames):
+                    raise ValueError(
+                        "The output catalog of at least one of your measurement functions does not"
+                        "contain the 'dx' and 'dy' columns which are mandatory for a single"
+                        "survey study."
+                    )
+
+            if isinstance(batch["blend_images"], dict):
+                if not ("ra" in out["catalog"].colnames and "dec" in out["catalog"].colnames):
+                    raise ValueError(
+                        "The output catalog of at least one of your measurement functions does not"
+                        "contain the 'ra' and 'dec' columns which are mandatory for a"
+                        "multi-resolution study."
+                    )
+
+            for key in ["deblended_images", "segmentation"]:
+                if key in out and out[key] is not None:
+                    if not isinstance(out[key], np.ndarray):
+                        raise TypeError(
+                            f"The output '{key}' of at least one of your measurement"
+                            f"functions is not a numpy array."
+                        )
+                    if not out[key].shape[-2:] == batch["blend_images"].shape[-2:]:
+                        raise ValueError(
+                            f"The shapes of the blended images in your {key} don't"
+                            f"match for at least one your measurement functions."
+                        )
+
+            out = {k: out.get(k, None) for k in self.measure_params}
+            output.append(out)
+        return output
 
     def __next__(self):
         """
         Returns:
-            draw_blend_generator output, deblender output and measurement output.
+            draw_blend_generator output from `__next__` method.
+            measurement output: List of len(batch_size), where each element is a list of
+                                len(measure_functions) corresponding to the measurements made by
+                                each function on each element of the batch. If only one
+                                measure_function was passed in, then each sub-list is converted
+                                to a single element.
         """
 
         blend_output = next(self.draw_blend_generator)
-        deblend_results = {}
-        measured_results = {}
         input_args = [(blend_output, i) for i in range(self.batch_size)]
```

Comment:
I guess when you write such list comprehension, `blend_output` is copied multiple times, which is quite inefficient. 
Instead, a generator would be much better.
```suggestion
        input_args = ((blend_output, i) for i in range(self.batch_size))
```